velox/docs/functions/spark/array.rst

```diff
@@ -24,6 +24,18 @@ Array Functions
         SELECT array_append(array(1, 2, 3), 2); -- [1, 2, 3, 2]
         SELECT array_append(array(1, 2, 3), NULL); -- [1, 2, 3, NULL]
 
+.. spark:function:: array_compact(array(E) x) -> array(E)
+
+    Removes all null elements from array ``x``. Returns NULL as result if array ``x`` is NULL.
+    If array ``x`` is empty array, returns empty array. If all elements in array ``x`` are NULL,
+    returns empty array. ::
+
+        SELECT array_compact(array(1, 2, NULL, 3)); -- [1, 2, 3]
+        SELECT array_compact(array()); -- []
+        SELECT array_compact(array(NULL)); -- []
+        SELECT array_compact(NULL); -- NULL
+        SELECT array_compact(array(1, 2), NULL, array(NULL, 3, 4)); -- [[1, 2], [NULL, 3, 4]]
```

Comment:
Met the below exception for this sql. It appears one top-level array is needed.

> org.apache.spark.sql.catalyst.ExtendedAnalysisException: [DATATYPE_MISMATCH.UNEXPECTED_INPUT_TYPE] Cannot resolve "array_compact(named_struct(col1, array(1, 2), col2, NULL, col3, array(NULL, 3, 4)))" due to data type mismatch: Parameter 1 requires the "ARRAY" type, however "named_struct(col1, array(1, 2), col2, NULL, col3, array(NULL, 3, 4))" has the type "STRUCT<col1: ARRAY<INT>, col2: VOID, col3: ARRAY<INT>>".;

Reply:
my bad, thanks